Framework/Algorithms/src/CreateSampleWorkspace.cpp

```diff
@@ -309,6 +320,36 @@ MatrixWorkspace_sptr CreateSampleWorkspace::createHistogramWorkspace(
   return create<Workspace2D>(inst, indexInfo, Histogram(x, y));
 }
 
+/** Create scanning histogram workspace
+ */
+MatrixWorkspace_sptr CreateSampleWorkspace::createScanningWorkspace(
+    int numPixels, int numBins, int numMonitors, double x0, double binDelta,
+    Geometry::Instrument_sptr inst, const std::string &functionString,
+    bool isRandom, int numScanPoints) {
+  auto builder =
+      ScanningWorkspaceBuilder(numPixels + numMonitors, numScanPoints, numBins);
+  builder.setInstrument(inst);
+
+  auto angles = std::vector<double>();
+  auto timeRanges = std::vector<double>();
+  for (int i = 0; i < numScanPoints; ++i) {
+    angles.push_back(double(i));
+    timeRanges.push_back(double(i + 1));
+  }
+
+  builder.setTimeRanges(Kernel::DateAndTime(0), timeRanges);
```

Comment:
I think it would be good to make the setters take r-value references. That way we can just move these temporaries rather than copying them.